packages/flutter/test/widgets/sliver_floating_header_test.dart

```diff
@@ -234,4 +234,87 @@ void main() {
     await tester.pump(const Duration(milliseconds: 500));
     expect(getHeaderRect(), const Rect.fromLTRB(0, 0, 800, 200));
   });
+
+  testWidgets('SliverFloatingHeader snapMode parameter', (WidgetTester tester) async {
+    Widget buildFrame(FloatingHeaderSnapMode snapMode) {
+      return MaterialApp(
+        home: Scaffold(
+          body: CustomScrollView(
+            slivers: <Widget>[
+              SliverFloatingHeader(
+                snapMode: snapMode,
+                child: const SizedBox(height: 200, child: Text('header')),
+              ),
+              SliverList(
+                delegate: SliverChildBuilderDelegate(
+                  (BuildContext context, int index) {
+                    return SizedBox(height: 100, child: Text('item $index'));
+                  },
+                  childCount: 100,
+                ),
+              ),
+            ],
+          ),
+        ),
+      );
+    }
+
+    Rect getHeaderRect() => tester.getRect(find.text('header'));
+    double getItem0Y() => tester.getRect(find.text('item 0')).topLeft.dy;
+
+    Future<void> scroll(Offset offset) async {
+      return tester.timedDrag(find.byType(CustomScrollView), offset, const Duration(milliseconds: 500));
+    }
+
+    // FloatingHeaderSnapMode.overlay
+    {
+      await tester.pumpWidget(buildFrame(FloatingHeaderSnapMode.overlay));
+      await tester.pumpAndSettle();
+      expect(getHeaderRect(), const Rect.fromLTRB(0, 0, 800, 200));
+      expect(getItem0Y(), 200);
+
+      await scroll(const Offset(0, -200));
+      await tester.pumpAndSettle();
+      expect(find.text('header'), findsNothing);
+      final double item0StartY = getItem0Y();
+      expect(item0StartY, lessThan(0));
```

Comment:
Should this be equal to 0?

Reply:
Big scrolls like this - offset.y == 200 - have a ballistic component, which is why we end up below 0. I added comments to the test explaining this and the fact that small timedDrag scrolls don't go ballistic.